client/components/foldable-card/index.jsx

```diff
@@ -72,86 +91,79 @@ class FoldableCard extends Component {
-	}
-
-	getActionButton() {
-		if ( this.state.expanded ) {
-			return this.props.actionButtonExpanded || this.props.actionButton;
-		}
-		return this.props.actionButton;
-	}
-
-	renderActionButton() {
-		const clickAction = ! this.props.clickableHeader ? this.getClickAction() : null;
-		if ( this.props.actionButton ) {
-			return (
-				<div className="foldable-card__action" onClick={ clickAction }>
-					{ this.getActionButton() }
-				</div>
-			);
-		}
-		if ( this.props.children ) {
-			const iconSize = 24;
-			const screenReaderText = this.props.screenReaderText || this.props.translate( 'More' );
-			return (
-				<button
-					disabled={ this.props.disabled }
-					type="button"
-					className="foldable-card__action foldable-card__expand"
-					onClick={ clickAction }
-				>
-					<ScreenReaderText>{ screenReaderText }</ScreenReaderText>
-					<Gridicon icon={ this.props.icon } size={ iconSize } />
-				</button>
-			);
-		}
-	}
-
-	renderContent() {
-		return <div className="foldable-card__content">{ this.props.children }</div>;
-	}
-
-	renderHeader() {
-		const summary = this.props.summary ? (
-			<span className="foldable-card__summary">{ this.props.summary } </span>
-		) : null;
-		const expandedSummary = this.props.expandedSummary ? (
-			<span className="foldable-card__summary-expanded">{ this.props.expandedSummary } </span>
-		) : null;
-		const headerClickAction = this.props.clickableHeader ? this.getClickAction() : null;
-		const headerClasses = classNames( 'foldable-card__header', {
-			'is-clickable': !! this.props.clickableHeader,
-			'has-border': !! this.props.summary,
-		} );
-		return (
-			<div className={ headerClasses } onClick={ headerClickAction }>
-				<span className="foldable-card__main">{ this.props.header } </span>
-				<span className="foldable-card__secondary">
-					{ summary }
-					{ expandedSummary }
-					{ this.renderActionButton() }
-				</span>
-			</div>
-		);
-	}
-
 	render() {
-		const Container = this.props.compact ? CompactCard : Card;
-		const itemSiteClasses = classNames( 'foldable-card', this.props.className, {
-			'is-disabled': !! this.props.disabled,
+		const {
+			actionButton,
+			actionButtonExpanded,
+			className,
+			clickableHeader,
+			compact,
+			disabled,
+			expandedSummary,
+			header,
+			icon,
+			screenReaderText,
+			summary,
+			translate,
+		} = this.props;
+
+		const Container = compact ? CompactCard : Card;
+
+		const itemSiteClasses = classNames( 'foldable-card', className, {
+			'is-disabled': !! disabled,
 			'is-expanded': !! this.state.expanded,
-			'has-expanded-summary': !! this.props.expandedSummary,
+			'has-expanded-summary': !! expandedSummary,
+		} );
+
+		const headerClasses = classNames( 'foldable-card__header', {
+			'is-clickable': !! clickableHeader,
+			'has-border': !! summary,
 		} );
 
 		return (
 			<Container className={ itemSiteClasses }>
-				{ this.renderHeader() }
-				{ this.state.expanded && this.renderContent() }
+				<div
```

Comment:
We can actually ignore the a11y linter here and just have the `onClick`, since the expander arrow should be there for keyboard users.

```jsx
{ /* eslint-disable jsx-a11y/click-events-have-key-events */ }
{ /* eslint-disable jsx-a11y/no-static-element-interactions */ }
<div className={ headerClasses } onClick={ this.clickHeader }>
	{ /* eslint-enable */ }
```